streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamsPartitionAssignor.java

```diff
@@ -768,15 +775,21 @@ private void assignTasksToClients(final Cluster fullMetadata,
 
         final Optional<org.apache.kafka.streams.processor.assignment.TaskAssignor> userTaskAssignor =
             userTaskAssignorSupplier.get();
+        UserTaskAssignmentListener userTaskAssignmentListener = (GroupAssignment assignment, GroupSubscription subscription) -> { };
         if (userTaskAssignor.isPresent()) {
             final ApplicationState applicationState = buildApplicationState(
                 taskManager.topologyMetadata(),
                 clientMetadataMap,
                 topicGroups,
                 fullMetadata
             );
-            final TaskAssignment taskAssignment = userTaskAssignor.get().assign(applicationState);
+            final org.apache.kafka.streams.processor.assignment.TaskAssignor assignor = userTaskAssignor.get();
+            final TaskAssignment taskAssignment = assignor.assign(applicationState);
             processStreamsPartitionAssignment(clientMetadataMap, taskAssignment);
+            final AssignmentError assignmentError = validateTaskAssignment(applicationState, taskAssignment);
+            userTaskAssignmentListener = (GroupAssignment assignment, GroupSubscription subscription) -> {
+                assignor.onAssignmentComputed(assignment, subscription, assignmentError);
+            };
```

Comment:
nit: Streams coding style tip, use simplified lambdas, ie

```suggestion
            userTaskAssignmentListener = (assignment, subscription) -> assignor.onAssignmentComputed(assignment, subscription, assignmentError);
```

(I'm actually surprised checkstyle isn't complaining about this. Your IDE is probably at least telling you to simplify it I would guess)